.github/workflows/ci.yml

```
@@ -25,76 +25,103 @@ env:
   MAVEN_CACHE_PATH: ~/.m2/repository
 
 jobs:
-  build-and-test:
+  verify-cms-docker-image:
     runs-on: ubuntu-latest
-
     steps:
-      - uses: actions/checkout@v3
-
-      - name: Set up JDK 17
-        uses: actions/setup-java@v3
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-java@v3
         with:
           java-version: '17'
           distribution: 'adopt'
-
       - uses: actions/cache/restore@v3
         with:
           path: ${{ env.MAVEN_CACHE_PATH }}
           key: ${{ env.MAVEN_CACHE_VERSION }}-${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ env.MAVEN_CACHE_VERSION }}-${{ runner.os }}-maven
+      - uses: docker/setup-qemu-action@v2
+      - uses: docker/setup-buildx-action@v2
+      - name: Build Docker image
+        run: ./mvnw --batch-mode install -DskipITs       # we build Docker image within install phase so updating maven cache is not allowed here
 
-      - name: Set up QEMU # https://github.com/docker/buildx/issues/499
-        uses: docker/setup-qemu-action@v2
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
-
-      - name: Build and run e2e tests
+  verify-cms-feature-model-e2e:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+      - uses: actions/cache/restore@v3
+        with:
+          path: ${{ env.MAVEN_CACHE_PATH }}
+          key: ${{ env.MAVEN_CACHE_VERSION }}-${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ env.MAVEN_CACHE_VERSION }}-${{ runner.os }}-maven
+      - name: Verify Starter CMS Feature model with integration and E2E tests
         run: ./mvnw --batch-mode clean verify -P e2e-functional
-
       - uses: actions/cache/save@v3
         if: always()
         with:
           path: ${{ env.MAVEN_CACHE_PATH }}
           key: ${{ env.MAVEN_CACHE_VERSION }}-${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
 
-      - name: Switch to base branch
-        uses: actions/checkout@v3
+  prepare-data-for-visual-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```

```suggestion
      - name: Check out base branch
        uses: actions/checkout@v4
```

Name would be better than comment in this case. And this is not always `main` branch, this is base, the target of PR.